packages/typescript/src/index.ts

```diff
@@ -19,17 +21,22 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
   const {
     cacheDir,
     compilerOptions,
-    filter,
     transformers,
     tsconfig,
     tslib,
+    include,
+    exclude,
+    filterRoot,
     typescript: ts
   } = getPluginOptions(options);
   const tsCache = new TSCache(cacheDir);
   const emittedFiles = new Map<string, string>();
   const watchProgramHelper = new WatchProgramHelper();
 
   const parsedOptions = parseTypescriptConfig(ts, tsconfig, compilerOptions);
+  const filter = createFilter(include || ['*.ts+(|x)', '**/*.ts+(|x)'], exclude, {
+    resolve: filterRoot ?? compilerOptions.rootDir
```

Comment:
Can this be handled inside `getPluginOptions`? It looks like the needed variables (`include`, `exclude`, `filterRoot`, `compilerOptions`) are available in that scope too.

Reply:
Forgot to change to `parsedOptions` in the oss repo, which isn't available in `getPluginOptions`. It could be refactored to pull the parse config logic into `getPluginOptions`, but that function is only used once with no defined return interface and seems to make sense to keep that single purpose.  Trying to keep changes minimal, it seems appropriate to keep separate the options normalization and the utility instantiations.